core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3497,24 +3503,29 @@ private int getWriterCount(Session session, WriterScalingOptions connectorScalin
                 return 1;
             }
 
+            int maxWritersBasedOnMemory = getMaxPartitionWritersBasedOnMemory(session);
             if (partitioningScheme.isPresent()) {
                 // The default value of partitioned writer count is 32 which is high enough to use it
                 // for both cases when scaling is enabled or not. Additionally, it doesn't lead to too many
                 // small files since when scaling is disabled only single writer will handle a single partition.
+                int partitionedWriterCount = getTaskWriterCount(session);
```

Comment:
Shouldn't it be `getTaskPartitionedWriterCount` as it was `getTaskPartitionedWriterCount(session)` before?